etna/transforms/detrend.py

```diff
@@ -1,41 +1,45 @@
+from typing import Optional
+
 import pandas as pd
+from sklearn.base import RegressorMixin
 from sklearn.linear_model import LinearRegression
 from sklearn.linear_model import TheilSenRegressor
 
 from etna.transforms.base import PerSegmentWrapper
 from etna.transforms.base import Transform
 
 
-class LinearTrendBaseTransform(Transform):
+class OneSegmentLinearTrendBaseTransform(Transform):
     """LinearTrendBaseTransform is a base class that implements trend subtraction and reconstruction feature."""
 
     is_categorical = False
     is_static = False
 
-    def __init__(self, in_column: str, regressor=None):
+    def __init__(self, in_column: str, regressor: Optional[RegressorMixin] = None):
         # TODO: add inplace arg
         """
-        Create instance of LinearTrendFeature.
+        Create instance of OneSegmentLinearTrendBaseTransform.
 
         Parameters
         ----------
-        regressor: instance of sklearn regressors to predict trend
+        regressor:
+            instance of sklearn RegressorMixin to predict trend
         """
         self._linear_model = regressor
         self.in_column = in_column
 
-    def fit(self, df: pd.DataFrame):
+    def fit(self, df: pd.DataFrame) -> "OneSegmentLinearTrendBaseTransform":
```

Comment:
from __future__ import annotations, doesn't it?

Reply:
As I can see, python3.6 doesn't support this feature but we try to support python3.6.